### PR TITLE
Fix+enable parallel build support on mono

### PR DESCRIPTION
**Pull request by radical**

### CODE REVIEW

**Comment by radical** [comment]
@rainersigwald any other tests that I should re-enable and test with this?

**Comment by radical** [comment]
Corresponding upstream PR: https://github.com/microsoft/msbuild/pull/4995

File: src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs [hunk 1]
```diff
@@ -441,10 +441,14 @@ private int LaunchNode(string msbuildLocation, string commandLineArgs)
 
             string exeName = msbuildLocation;
 
-#if RUNTIME_TYPE_NETCORE
-            // Run the child process with the same host as the currently-running process.
-            exeName = GetCurrentHost();
-            commandLineArgs = "\"" + msbuildLocation + "\" " + commandLineArgs;
+#if RUNTIME_TYPE_NETCORE || MONO
```
Comment (marek-safar) [inline src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs:444]: Why is this needed when it's disabled for Mono

Comment (radical) [reply src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs:444]: Just in preparation for being able to remove `#define MONO` in the future and have unified code paths.

**Review by kg — APPROVED** [review]
